docs/http-api-integration.md

```diff
@@ -0,0 +1,106 @@
+provide the needed additional constraints that address the previously listed "techniques" that RESTful API does not address. 
+There are many features that JSON API and GraphQl are capable of, but we are only focusing on the ones that achieve the goal of reducing the size and number of API requests that go between the client and server.
+
+## Adding Constraints through Specifications
+
+### JSON:API
+**JSON:API** has features such as **Sparse Fields**, **Compound Documents**, **Pagination**, and **Caching**.
+
+#### Sparse Fieldset
+This is the bread and butter that allows the request to specify the fields needed.
+There are more examples on Sparse Fieldsets at [jsonapi.org](https://jsonapi.org/examples/#sparse-fieldsets)
+```
+GET /books?fields[books]=author,themes 
+```
+
+#### Compound Documents
+This feature allows for including related resources with the initial request
+```
+GET /books?fields[books]=author,themes,&include=chapter1/title 
+```
+
+#### Caching
+JSON:API takes advantage of the caching ability that HTTP has built in. This reduces the need to return data that has not changed, saving bandwidth.
+
+#### Pagination
+JSON:API handles Pagination with a standard that `links` such as: `first`, `last`, `next`, and `previous` will return subnets of sort information.
+https://jsonapi.org/format/#fetching-pagination
+
+---
+
+### GraphQL
+**GraphQL** can also reduce the fields returned by specifying them in the GraphQL Query. Keep in mind, as its name implies GraphQL is predominantly a Query Language
+
+#### Sparse Fieldsets
+```
+GET /graphql
+books(title: Hobbit) {
+  author
+  themes
+}
+
+response
+"books": {
+  author: J.R.R. Tolkien
+  themes: fantasy, fiction
+}
+```
+
+#### Caching
+GraphQL utilizes HTTP Caching and identifies when objects with globally uniquie identifiers are called again.
+https://graphql.org/learn/caching/#gatsby-focus-wrapper
+
+#### Pagination
+GraphQL suggests various way to paginate returning data by requesting a set such as `first:10 offset:20` which is the equivalent to saying "second page of 10 items". 
+Another example is `first:5 offset:$itemId`, equivalent of saying "first 5 items after the ID of this item".
+https://graphql.org/learn/pagination/
+
+## Architecture
+
+### RESTful
+RESTful API uses HTTP verbs (Post, Get, Patch, etc...) in order to manipulate data. 
+Because of this, designing a RESTful system requires creating controllers that enact actions (e.g. [POST] /user, [GET] /user/address, [PATCH] /user/age) 
+JSON:API does not deviate from the Controller-Model structure that RESTful uses. 
+
+![Alt text](images/restful-api-diagram.png?raw=true)
+
+
+### GraphQL
+In GraphQL, Queries are sent with HTTP Get to retrieve resources from a database. When modifying data, it is NOT recommended to use HTTP Get. See documentation on Mutations at this [link](https://graphql.org/learn/queries/#mutations). 
```

Comment:
I don't know how this sentence connects to the above or how it is about "Architecture" Perhaps you mean "Workflow" or something similar.

Reply:
I don't think I add anything useful with this Architecture section. On top of that, the GraphQL diagram doesnt reflect how I describe the usage of GET and POST. I'm just gonna remove the whole section and end with "Final Thoughts"